core/src/main/java/com/datastax/oss/driver/api/core/type/codec/ExtraTypeCodecs.java

```diff
@@ -0,0 +1,482 @@
+   *
+   * @see #ZONED_TIMESTAMP_SYSTEM
+   * @see #ZONED_TIMESTAMP_UTC
+   * @see #zonedTimestampAt(ZoneId)
+   */
+  public static final TypeCodec<ZonedDateTime> ZONED_TIMESTAMP_PERSISTED =
+      new PersistentZonedTimestampCodec();
+
+  /**
+   * A codec that maps CQL type {@code timestamp} to Java's {@link LocalDateTime}, using the
+   * system's default time zone.
+   *
+   * <p>This codec uses the system's {@linkplain ZoneId#systemDefault() default time zone} as its
+   * source of time zone information when encoding or decoding. If you need a different time zone,
+   * consider using other constants in this class, or call {@link #localTimestampAt(ZoneId)}
+   * instead.
+   *
+   * <p>Note that CQL type {@code timestamp} does not store any time zone; this codec is provided
+   * merely as a convenience for users that need to deal with local date-times in their
+   * applications.
+   *
+   * @see #LOCAL_TIMESTAMP_UTC
+   * @see #localTimestampAt(ZoneId)
+   */
+  public static final TypeCodec<LocalDateTime> LOCAL_TIMESTAMP_SYSTEM = new LocalTimestampCodec();
+
+  /**
+   * A codec that maps CQL type {@code timestamp} to Java's {@link LocalDateTime}, using the UTC
+   * time zone.
+   *
+   * <p>This codec uses {@link ZoneOffset#UTC} as its source of time zone information when encoding
+   * or decoding. If you need a different time zone, consider using other constants in this class,
+   * or call {@link #localTimestampAt(ZoneId)} instead.
+   *
+   * <p>Note that CQL type {@code timestamp} does not store any time zone; this codec is provided
+   * merely as a convenience for users that need to deal with local date-times in their
+   * applications.
+   *
+   * @see #LOCAL_TIMESTAMP_SYSTEM
+   * @see #localTimestampAt(ZoneId)
+   */
+  public static final TypeCodec<LocalDateTime> LOCAL_TIMESTAMP_UTC =
+      new LocalTimestampCodec(ZoneOffset.UTC);
+
+  /**
+   * A codec that maps CQL type {@code blob} to Java's {@code byte[]}.
+   *
+   * <p>If you are looking for a codec mapping CQL type {@code blob} to the Java type {@link
+   * ByteBuffer}, you should use {@link TypeCodecs#BLOB} instead.
+   *
+   * <p>If you are looking for a codec mapping CQL type {@code list<tinyint} to the Java type {@code
+   * byte[]}, you should use {@link #BYTE_LIST_TO_ARRAY} instead.
+   *
+   * @see TypeCodecs#BLOB
+   * @see #BYTE_LIST_TO_ARRAY
+   */
+  public static final TypeCodec<byte[]> BLOB_TO_ARRAY = new SimpleBlobCodec();
+
+  /**
+   * A codec that maps CQL type {@code list<boolean>} to Java's {@code boolean[]}.
+   *
+   * <p>Note that this codec is designed for performance and converts CQL lists <em>directly</em> to
+   * {@code boolean[]}, thus avoiding any unnecessary boxing and unboxing of Java primitive {@code
+   * boolean} values; it also instantiates arrays without the need for an intermediary Java {@code
+   * List} object.
+   */
+  public static final TypeCodec<boolean[]> BOOLEAN_ARRAY = new BooleanListToArrayCodec();
```

Comment:
@olim7t I'm overall 👍 on the new class but why did you change the name of this field and not the others below?

Reply:
🤔  I thought I had (hence the correct name in the manual). Will do now.